apps/snitch_core/priv/repo/migrations/20180619142113_embed_order_address.exs

```diff
@@ -0,0 +1,10 @@
+defmodule Snitch.Repo.Migrations.EmbedOrderAddress do
+  use Ecto.Migration
+
+  def change do
+    alter table(:snitch_orders) do
+      add :billing_address, :map
```

Comment:
Please update the Order schema as well.
```elixir
embeds_one(:billing_address, Address, one_replace: :something_sane_here)
```
and remove the `*_address_id` columns from the table.

Reply:
## TODO
I updated them, but need to fix the **test cases**  in  various files in `shipment_test.exs` and `transitions_test.exs`. For that need your assistance.
Here, we need to update test cases where every address is associated.

I usually see the common setup line `import Snitch. Factory` is that where you write all the `setup` definitions?

## Declaration
I have to visit more files to update myself, which exactly I don't know the flow, If I do changes, that may reflect some other test cases.

## Assistance From you @oyeb 
 1. I did not find one of the definitions `setup: user_with_address`. Can you help me with that?
2. Can you update the test cases if possible according to changed migration after deleting `billing_address_id` and `shipping_address_id`? 

## Code Changes
Please refer the latest comment [removed address_ids and updated order schema](https://github.com/aviabird/snitch/pull/104/commits/643dc1852a5c07a57f5b1ede5ae843fbb99d4694)

Reply:
@blackode You can update the test cases where ever required. @oyeb Can tell you where/how have he added those. But try out yourself, instead of waiting for him.

Reply:
@pkrawat1 As you suggested, I updated the test cases by traveling each and every file. 

## Day end Experience
It was a nice experience. I got chance to read each and every test case. I figured out the files associated with testing which helps in future to write own test cases without assistance.

## The only problem
I usually encounter with  the following error

```
  ** (CaseClauseError) no case clause matching: {:id, :id, :id}
``` 
Did you guys experience this before? If yes, kindly share your knowledge on this.
This error comes whenever I come across the line `Repo.update`, which leads to failure of 5 test cases that are dealing with `Repo.update`. 

## Error References 
I tried to debug this in the project LOC itself.  I felt nothing strange. So, I google it and found the following helplines.
https://github.com/thoughtbot/ex_machina/issues/240
It is saying, using `ex_machina` with `ecto ~> 2.2` causing this error. 
After referring above URL, I questioned myself; why it worked before but not now?

## Code Changes
Please refer to latest comment :point_down: 
[test cases updated for embedding address in order](https://github.com/aviabird/snitch/pull/104/commits/bb88e6011bfbe3eddd54b1eb48fa30fe74ac0ab4). 

Pull the code and test in your machine as well. Let me know your experience in comments :).